File: Kata 6/IQ Test/IQ Test.cs
/*

Assert.AreEqual(3, IQ.Test("2 4 7 8 10"));
Assert.AreEqual(1, IQ.Test("1 2 2"));
*/
using System;
using System.Linq;
using System.Collections.Generic;

public class IQ
    {	
				
		public static int Test1(string numbers)
        { 
            var ints = numbers.Split(' ').Select(int.Parse).ToList();
            var unique = ints.GroupBy(x => x % 2).OrderBy(y => y.Count()).First().First();
            return ints.FindIndex(c => c == unique) + 1;
        }
		
		 public static int Test2(string numbers)
		{ 
			var nums = numbers.Split(' ').Select((s, i) => new { Value = int.Parse(s), Index = i });
			var even = nums.Where(x => x.Value % 2 == 0);
			var odd = nums.Where(x => x.Value % 2 == 1);
			return even.Count() == 1 ? even.First().Index + 1: odd.First().Index + 1; 
		}
    }

File: Kata 6/IQ Test/IQ Test.py
'''

Assert.AreEqual(3, IQ.Test("2 4 7 8 10"));
Assert.AreEqual(1, IQ.Test("1 2 2"));
'''



def iq_test(numbers):
    	reslut=[ int(i)%2 for i in numbers.split(' ')]
	if 	(reslut.count(0)==1):
		return reslut.index(0)+1
	else:	
		return reslut.index(1)+1

def iq_test2(numbers):
    e = [int(i) % 2 == 0 for i in numbers.split()]
    return e.index(True) + 1 if e.count(True) == 1 else e.index(False) + 1
   
print iq_test2("2 4 7 8 10")